connectors/sources/abs.py

```diff
@@ -0,0 +1,219 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+"""Azure Blob Storage source module responsible to fetch documents from Azure Blob Storage"""
+
+import os
+from functools import partial
+
+from azure.storage.blob.aio import BlobClient, BlobServiceClient, ContainerClient
+
+from connectors.logger import logger
+from connectors.source import BaseDataSource
+from connectors.utils import TIKA_SUPPORTED_FILETYPES, get_base64_value
+
+BLOB_SCHEMA = {
+    "title": "name",
+    "tier": "blob_tier",
+    "size": "size",
+    "container": "container",
+}
+DEFAULT_CONTENT_EXTRACTION = True
+DEFAULT_FILE_SIZE_LIMIT = 10485760
+DEFAULT_RETRY_COUNT = 3
+
+
+class AzureBlobStorageDataSource(BaseDataSource):
+    """Class to fetch documents from Azure Blob Storage"""
+
+    def __init__(self, connector):
+        """Setup the connection to the azure base client
+
+        Args:
+            connector (BYOConnector): Object of the BYOConnector class
+        """
+        super().__init__(connector=connector)
+        self.connection_string = self.configuration["connection_string"]
+        self.enable_content_extraction = self.configuration.get(
+            "enable_content_extraction", DEFAULT_CONTENT_EXTRACTION
+        )
+        self.retry_count = int(
+            self.configuration.get("retry_count", DEFAULT_RETRY_COUNT)
+        )
+
+    @classmethod
+    def get_default_configuration(cls):
+        """Get the default configuration for Azure Blob Storage
+
+        Returns:
+            dictionary: Default configuration
+        """
+        return {
+            "connection_string": {
+                "value": "DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;BlobEndpoint=http://127.0.0.1:10000/devstoreaccount1;",
```

Comment:
can we move the key in a constant at the top?

Reply:
Yes, we can replace connection string with `AccountName` and `AccountKey` but we required to form an account url with the help of `AccountName`.
Also same like GCS and S3 we need to set a variable in ftest `RUNNING_FTEST` to bifurcate the local account url for e2e and the cloud account url to run the connector. 

So, Should we procced further with `AccountName` & `AccountKey` ?

Reply:
> we required to form an account url with the help of AccountName

This can be when by composing the values from the config when you need to create a connection, right?

> So, Should we procced further with AccountName & AccountKey ?

I think so, what I think could be done is replace `connection_string` with:

- account_name
- account_key
- blob_endpoint
- database_name maybe?

 and infer the value of `DefaultEndPointsProtocol`, so we do:


```
proto = extract_protocol(blob_endpoint)
connection_string = f'DefaultEndpointsProtocol={proto};AccountName={account_name};AccountKey={account_key};BlobEndpoint={blob_endpoint}/{database};'
```